src/Admin/Startup.cs

```diff
@@ -82,6 +82,8 @@ public void ConfigureServices(IServiceCollection services)
             });
         }
 
+        services.AddOosServices();
```

Comment:
Yes, this is more of a problem than I first thought. `OrganizationService` is a base service that is used everywhere in OSS code (such as Identity), it shouldn't have any proprietary code in it at all. Even using a noop service means that we need to import the `IServiceAccountRepository` interface from BL code into OSS code. And generally it extends the OSS/Commercial distinction to additional projects, I'm not comfortable doing that, especially when this particular method is only going to be used in the cloud anyway.

I'm not sure how to solve this, my thought is to do a lift & shift of `OrganizationService.UpgradePlanAsync` out of the service and to its own command. (@cyprain-okeke did this initially but it didn't seem necessary then, but it might be more necessary now.) That would get the dependency out of `OrganizationService`. The controller endpoint may need to be separate as well - I'm not exactly sure how the dependencies are resolved in this case. Please investigate this and discuss it with one of the tech leads if needed, and we can sync up on it next week.

If we are going to lift & shift any large amount of code, let's do it in a separate PR so that we don't mess up the diff here.

Reply:
I think that temporarily we could add the mock `NoopServiceAccountRepository` directly and create a ticket to `OrganizationService.UpgradePlanAsync` out into its own command refactor.

I don't like temporary solutions - they tend to be permanent - but we have time constraints.

Reply:
I have. a call with Justin today but he suggested adding services.AddSecretsManagerEfRepositories();  to Identity/startup and Admin/Startup but thinks that Separating UpgradePlanAsync code from OrganizationService is the best solution if we can quickly achieve it. I  want to create a task to move UpgradePlanAsync code from OrganizationServices, to track it.